build.rs

```diff
@@ -264,13 +265,28 @@ fn main() {
         }
     }
 
-    if std::path::Path::new("/dev/sgx_enclave").exists() {
+    let host_can_test_sgx = if std::path::Path::new("/dev/sgx_enclave").exists() {
         // Not expected to fail, as the file exists.
         let metadata = fs::metadata("/dev/sgx_enclave").unwrap();
         let file_type = metadata.file_type();
 
         if file_type.is_char_device() {
             println!("cargo:rustc-cfg=host_can_test_sgx");
+            true
```

Comment:
Can probably avoid the ifs and elses by replacing this `true` with `if file_type.is_socket() && std::path::Path::new(AESM_SOCKET).exists() {`

Reply:
There was actually a bug. I checked file_type() for the device file, not socket.

Reply:
This should be correct now:
```rust
    let host_can_test_sgx = std::path::Path::new("/dev/sgx_enclave").exists()
        && fs::metadata("/dev/sgx_enclave")
            .unwrap()
            .file_type()
            .is_char_device();

    if host_can_test_sgx {
        println!("cargo:rustc-cfg=host_can_test_sgx");

        if std::path::Path::new(AESM_SOCKET).exists()
            && fs::metadata(AESM_SOCKET).unwrap().file_type().is_socket()
        {
            println!("cargo:rustc-cfg=host_can_test_attestation");
        }
    }

 ```